include/swift/Runtime/Config.h

```diff
@@ -38,7 +38,7 @@
 #ifndef SWIFT_USE_SWIFTCALL
 // Clang doesn't support mangling functions with the swiftcall attribute
 // on Windows and crashes during compilation: http://bugs.llvm.org/show_bug.cgi?id=32000
-#if (__has_attribute(swiftcall) || defined(__linux__)) && !defined(_WIN32)
+#if __has_attribute(swiftcall) && !defined(_WIN32)
```

Comment:
IRGen doesn't really do LLVM-IR CC lowering properly in the case where the native CC is not swiftcc; to do so, it would really need to export the original signature as C types, whip up a ForeignFunctionInfo, and follow that religiously even for native calls.  It mostly doesn't matter because it only affects compiler/runtime interoperation for the set of native function types actually used by the runtime, which is quite small and mostly just uses pointers and pointer-sized ints, i.e. types for which the swiftcc and C conventions prescribe the same IR lowering anyway.

I would really like us to get to the point of only supporting the runtime being built with a just-built clang.  Our language model depends on us being able to generate ABI-compliant code for the target using that clang anyway.  If we need to cut target-SDK dependencies (C++ libraries?) from the runtime to do that more easily, so be it.  If we can get to that point, then we can unconditionally use swiftcc in the runtime and let Clang make the decision about whether that actually maps to llvm::CallingConv::Swift.

Until we get to that point, there's probably some reasonable way to define an x-macro database that will (1) determine this correctly for building the runtime and (2) allow us to pick it up (or assert it) correctly in the compiler.